src/main/java/org/jabref/gui/maintable/MainTable.java

```diff
@@ -280,6 +298,34 @@ public void cut() {
         libraryTab.delete(StandardActions.CUT);
     }
 
+    private void scrollToRank(int delta) {
+        BibEntryTableViewModel selectedEntry = getSelectionModel().getSelectedItem();
+        if (selectedEntry == null) {
+            return;
+        }
+
+        List<BibEntryTableViewModel> firstEntryOfEachRank = new ArrayList<>(Collections.nCopies(FOURTH_RANK + 1, null));
+        for (BibEntryTableViewModel entry : getItems()) {
+            int rank = entry.searchRank().get();
+            if (firstEntryOfEachRank.get(rank) == null) {
+                firstEntryOfEachRank.set(rank, entry);
+            }
+            if (rank == FOURTH_RANK) {
+                break;
+            }
+        }
+
+        for (int i = selectedEntry.searchRank().get() + delta; i > 0 && i < firstEntryOfEachRank.size(); i += delta) {
+            BibEntryTableViewModel entry = firstEntryOfEachRank.get(i);
+            if (entry != null) {
+                getSelectionModel().clearSelection();
+                getSelectionModel().select(entry);
+                scrollTo(entry);
+                return;
+            }
+        }
+    }
```

Comment:
This seems not to be what the name of the method indicates.
I would guess "scroll to rank" means scroll to a defined rank. But the method scrolls to a delta in rank (whatever that means). Also the method seems way too complex. You create information in firstEntryOfEachRank you don't need in the end.
Look for the currently selected item, read the current rank, loop through the list of the first best matching entry and jump there.

Reply:
This method scrolls to the next or previous rank based on a delta of `1` or `-1`. I will rename the method to `scrollToAdjacentRank` or `scrollToNextOrPreviousRank`. 
If I create two separate methods, `scrollToNextRank` and `scrollToPreviousRank`, the code will be duplicated except for the sign used (`+` or `-`) while iterating.

I already use the `firstEntryOfEachRank` on line 319. 
I want to collect the first entry for each rank because there may be gaps between ranks, it's not guaranteed that `rank + 1` or `rank - 1` exists.

Reply:
You use firstEntryOfEachRank after looping through the entire entrylist until you have the fourth rank and creating a List of the first entry of each rank. this is not necessary. Just find out, which Rank you are looking for and then scroll to the first best you find.
Btw, what about if there is not entry of a rank?

Reply:
> Just find out, which Rank you are looking for and then scroll to the first best you find.

How do I find out which rank to scroll to? For example, if the current rank is 1 and there are no entries with ranks 2 or 3, but rank 4 exists, in this case I have to scroll to the first entry with rank 4.
How to do this with one iteration?



> You use firstEntryOfEachRank after looping through the entire entrylist until you have the fourth rank and creating a List of the first entry of each rank.

Instead of going through the whole list, I can use binary search to find the first entry of rank `i` and do this for all 4 ranks.

> Btw, what about if there is not entry of a rank?

If you're trying to move to the next rank and the current rank is the highest rank in the list, nothing will happen.

Reply:
Created two separate methods `scrollToNextRank`, `scrollToPreviousRank`, I think looks better now? [`108a39e` (#11510)](https://github.com/JabRef/jabref/pull/11510/commits/108a39e1bab82fbaa9652039b27ca63ad8e0c764)